src/BootstrapTable.js

```diff
@@ -283,6 +283,13 @@ class BootstrapTable extends Component {
           }
           newState.data = data;
         }
+
+        const { options: currentOptions } = this.props;
+        const sortName = options.sortName;
+        const sortOrder = options.sortOrder;
+        if (this.allowRemote(Const.REMOTE_SORT) && (currentOptions.sortName !== sortName || currentOptions.sortOrder !== sortOrder)) {
+          this.store.setSortInfo(sortOrder, options.sortName);
+        }
```

Comment:
This part only set the sort field and order to store object. but the data still not change by newest sort field and order.

My suggestion is if you change the sort field or order programmatically(not by clicking on header), then you are suppose to handle the sorting function in your program and then feeds newest data(sorted) into table

Because the remote mode will not handle the data change(sort, filter, pagination) internally.

Reply:
That is exactly my intention. The data is sorted on the backend and passed into the table via props. Sort order field(s) changed programatically sometimes (assume case when you need to reset the page to defaults).

So in other words, i assume that if table works in "remote" mode, then all the data logic is handled "remotely" and component is dumb (or almost dumb) and only renders stuff that is passed to it. This fix making sure that if you change sort order from outside, it get's synchronized back to internal variable inside of the table and renders properly.

Reply:
@aalbul got it, sorry I got some misunderstand before, but now is clear for me ;) thanks